_posts/2017-05-15-rust-at-two-years.md

```diff
@@ -0,0 +1,356 @@
+
+The [2017 Roadmap] goals have been great for focusing community efforts towards
+the most pressing issues facing Rust today. Of course we'd love for every
+aspect of Rust to improve all the time, but we don't have an infinite number of
+contributors with an infinite amount of time available yet!
+
+Let's check in on some of the initiatives in each of the goals in the roadmap.
+Take a look at each goal's tracking issue for even more initiatives than we're
+mentioning here.
+
+### [Rust should have a lower learning curve](https://github.com/rust-lang/rust-roadmap/issues/3)
+
+The second edition of [The Rust Programming Language Book] is one chapter shy
+of having its initial content complete. There's lots more editing to be done to
+get the book ready for publication in October, though. The print version is
+currently available for [preorder from No Starch], and the online version of
+the second edition has boarded [the beta train] and will be an option in the
+documentation shipped with Rust 1.18.0. Steve and I have gotten feedback that
+[the ownership chapter] especially is much improved and has helped people
+understand ownership related concepts better!
+
+[The Rust Programming Language Book]: https://github.com/rust-lang/book
+[preorder from No Starch]: https://www.nostarch.com/rust
+[the beta train]: https://doc.rust-lang.org/beta/book/
+[the ownership chapter]: https://doc.rust-lang.org/nightly/book/second-edition/ch04-00-understanding-ownership.html
+
+The [Language Ergonomics Initiative] is another part of the lower learning
+curve goal that has [a number of improvements][lei-tracker] in its pipeline.
+The language team is eager to mentor people (another goal!) who are interested
+in getting involved with moving these ergonomic improvement ideas forward by
+writing RFCs and working with the community to flesh out the details of how
+these improvements would work. Comment on the [tracking issue][lei-tracker] if
+you'd like to jump in.
+
+[Language Ergonomics Initiative]: https://blog.rust-lang.org/2017/03/02/lang-ergonomics.html
+[lei-tracker]: https://github.com/rust-lang/rust-roadmap/issues/17
+
+Also check out:
+
+- [The Rust Cookbook](https://github.com/brson/rust-cookbook)
+- [The new error format rolled out in Rust 1.12.0](https://github.com/rust-lang/rust/issues/35233)
+- [The question mark operator stabilized in Rust 1.13.0](https://github.com/rust-lang/rust/pull/31954)
+
+### [Rust should have a pleasant edit-compile-debug cycle](https://github.com/rust-lang/rust-roadmap/issues/1)
+
+Compiler improvements are ongoing. This year, [MIR became a default part of the
+compilation process](https://github.com/rust-lang/rust/pull/34096), which was a
+prerequisite to unlocking further improvements.
+
+Below is a table of [benchmarks] comparing the time it takes to compile a few
+crates and tests with Rust 1.8.0 (the stable release a year ago) to Rust 1.17.0
+(the stable release today). On average, compile times have improved by 5-10%,
```

Comment:
The 5-10% figure here is going to feel pretty underwhelming, given the focus we've put on compiler performance in the past. We should figure out a good way to frame that.